src/transactions/RestoreFootprintOpFrame.cpp

```diff
@@ -108,6 +108,12 @@ RestoreFootprintOpFrame::doApply(Application& app, AbstractLedgerTxn& ltx,
         // To maintain consistency with InvokeHostFunction, TTLEntry
         // writes come out of refundable fee, so only add entrySize
         metrics.mLedgerWriteByte += entrySize;
+        if (!validateContractLedgerEntry(lk, entrySize, sorobanConfig,
```

Comment:
I guess if we're doing this check everywhere, then we should also add it to extend ttl operation.